### ISSUE TITLE
Add telemetry event when the pty host becomes unresponsive

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#120185]
https://github.com/Microsoft/vscode/blob/f5146529508fcd1b94594ac7562f0d332edf86a5/src/vs/platform/terminal/node/ptyHostService.ts#L226-L226

cc @deepak1556

**Comment by bpasero** [comment]
Running a quick search over how many shared processes report "Unresponsive" (an electron event), I see around 6000 clients hitting this event for `1.55.2`. We still have no clear understanding how the shared process can become unresponsive overall, but we need to understand this going forward if the shared process is the home of the extension host and file services.

Having a better understanding of what happens in the terminal host in this case would be helpful 👍

**Comment by Tyriar** [comment]
I have a repro and have tried to mitigate it but it didn't end up working: https://github.com/microsoft/vscode/issues/121336

Was working on this yesterday but still having problems. The [root cause I fixed months ago](https://github.com/microsoft/node-pty/pull/415) but am blocked by workers not working only in the product build due to asar which I've been chatting to @deepak1556 about. It would be lovely to get this fixed for April but am waiting for @deepak1556 to free up as I've spent way too long on this with no success.

And yes I'm sick of adding mitigation on top of mitigation for a bug that I've already fixed ☹️

**Comment by Tyriar** [comment]
> Running a quick search over how many shared processes report "Unresponsive" (an electron event)

Also, are you sure this is related? Does this event fire when a _child process_ of a window is unresponsive? The pty host doesn't take down the shared process.

**Comment by bpasero** [comment]
@Tyriar yeah sorry, I should have made more clear: I cannot tell the unresponsiveness stems from the terminals host, just a speculation. I have now added more telemetry to learn what can cause this. So far we have tracked if the shared process window has been made visible by the user (rather unlikely) but now more importantly I track if the VSCode window is shutting down when this happens. According to Deepak, unresponsiveness can also fire when a shutdown takes more than 5s which maybe is the case for shared process.
